pepr/robustness/art_wrapper.py

```diff
@@ -0,0 +1,3106 @@
+        """
+        self._report_attack_configuration()
+        self._report_attack_results(save_path)
+
+    def _report_attack_configuration(self):
+        """Create subsubsection about the attack and data configuration."""
+        # Create tables for attack parameters and the data configuration.
+        tm = 0  # Specify target model
+
+        def gen_attack_pars_rows(table):
+            for key in self.pars_descriptors:
+                desc = self.pars_descriptors[key]
+                if key == "targeted":
+                    key = "_targeted"
+                elif key == "verbose":
+                    continue
+                value = str(self.art_attacks[tm].__dict__[key])
+
+                table.add_hline()
+                table.add_row([desc, value])
+
+        dc = self.data_conf
+        self.report_section.append(Subsubsection("Attack Details"))
+        with self.report_section.create(MiniPage()):
+            with self.report_section.create(MiniPage(width=r"0.49\textwidth")):
+                # -- Create table for the attack parameters.
+                self.report_section.append(Command("centering"))
+                with self.report_section.create(Tabular("|l|c|")) as tab_ap:
+                    tab_ap.add_hline()
+                    tab_ap.add_row(["Use true labels", self.use_labels])
+                    gen_attack_pars_rows(tab_ap)
+                    tab_ap.add_hline()
+                self.report_section.append(Command("captionsetup", "labelformat=empty"))
+                self.report_section.append(
+                    Command(
+                        "captionof",
+                        "table",
+                        extra_arguments="Attack parameters",
+                    )
+                )
+
+            with self.report_section.create(MiniPage(width=r"0.49\textwidth")):
+                # -- Create table for the data configuration
+                self.report_section.append(Command("centering"))
+                nr_targets, target_attack_set_size = dc[
+                    "attack_indices_per_target"
+                ].shape
+                with self.report_section.create(Tabular("|l|c|")) as tab_dc:
+                    tab_dc.add_hline()
+                    tab_dc.add_row(["Attacked target models", nr_targets])
+                    tab_dc.add_hline()
+                    tab_dc.add_row(
+                        ["Target model's attack sets size", target_attack_set_size]
+                    )
+                    tab_dc.add_hline()
+                self.report_section.append(Command("captionsetup", "labelformat=empty"))
+                self.report_section.append(
+                    Command(
+                        "captionof",
+                        "table",
+                        extra_arguments="Target and Data Configuration",
+                    )
+                )
+
+    def _report_attack_results(self, save_path):
```

Comment:
At the moment save_path isn't used, but I think you want to add assets?

Reply:
Yes, a histogram will be added in the next days.